server/src/main/java/org/opensearch/index/ShardIndexingPressureTracker.java

```diff
@@ -184,18 +212,34 @@ public long getLatencyInMillis() {
             return latencyInMillis.get();
         }
 
+        public long addLatencyInMillis(long latency) {
+            return latencyInMillis.addAndGet(latency);
+        }
+
         public long getLastSuccessfulRequestTimestamp() {
             return lastSuccessfulRequestTimestamp.get();
         }
 
+        public long updateLastSuccessfulRequestTimestamp(long timeStamp) {
+            return lastSuccessfulRequestTimestamp.getAndSet(timeStamp);
+        }
+
```

Comment:
This doesn't have to AtomicLong, just volatile should work

Reply:
++